src/merkle/sparse_path.rs

```diff
@@ -0,0 +1,109 @@
+use alloc::vec::Vec;
+use core::iter;
+
+use super::{EmptySubtreeRoots, MerklePath, RpoDigest};
+
+/// A different representation of [`MerklePath`] designed for memory efficiency for Merkle paths
+/// with empty nodes.
+///
+/// Empty nodes in the path are stored only as their position, represented with a bitmask. A
+/// maximum of 64 nodes in the path can be empty. The number of empty nodes has no effect on memory
+/// usage by this struct, but will incur overhead during iteration or conversion to a
+/// [`MerklePath`], for each empty node.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct SparseMerklePath<const DEPTH: u8> {
```

Comment:
I'm not sure about having `DEPTH` be a generic parameter. Sometimes, we may be dealing with the paths for which we don't know the depth (e.g., paths that come over the wire or are stored in the database).

Reply:
Note that, to my understanding, we will still need to know the depth at runtime, so that we can determine if a node is empty or not.